app/src/main/java/com/asfoundation/wallet/promo_code/bottom_sheet/entry/PromoCodeBottomSheetViewModel.kt

```diff
@@ -16,7 +19,7 @@ sealed class PromoCodeBottomSheetSideEffect : SideEffect {
 
 data class PromoCodeBottomSheetState(
   val promoCodeAsync: Async<PromoCodeResult> = Async.Uninitialized,
-  val submitClickAsync: Async<Unit> = Async.Uninitialized,
+  val submitClickAsync: Async<PromoCode> = Async.Uninitialized,
```

Comment:
here both async have different objectives:
- the promoCodeAsync is being used for deciding what to show when opening the screen based on the stored promo code
- the submitClickAsyc is for the direct result of inserting a new promo code (treating the success and possible errors).
It might be a good point to keep in mind when the view refactor is made, right now I'm not seeing a simple way to rely only in the first one for both situations.

Reply:
Got it